### PR TITLE
[test] build with cmake instead of automake

### PR DESCRIPTION
**Pull request by bukepo**
* Use cmake to build.
* Use `rcp` instead of `transceiver` simplicity.
* Add virtual time options for cmake.
* Enable readline support for posix apps.
* Use ninja instead of make.

### CODE REVIEW

File: src/posix/standalone.cmake [hunk 1]
```diff
@@ -26,14 +26,20 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
+set(OT_READLINE "readline" CACHE STRING "set readline library name(readline or edit)")
+find_library(READLINE ${OT_READLINE})
+string(TOUPPER ${OT_READLINE} READLINE_DEFINE)
```
Comment (bukepo) [inline src/posix/standalone.cmake]:
```suggestion
```

File: script/test [hunk 2]
```diff
@@ -192,35 +240,44 @@ do_package() {
...
-    export NODE_TYPE="${NODE_TYPE:-sim}"
-    export VERBOSE="${VERBOSE:-1}"
-    export VIRTUAL_TIME="${VIRTUAL_TIME:-1}"
-    export THREAD_VERSION="${THREAD_VERSION:-1.1}"
-
-    if [[ "${NODE_MODE}" = 'transceiver' ]] ; then
-        export RADIO_DEVICE="${OT_BUILDDIR}/${OT_SYSTEM_TRIPLET}/examples/apps/ncp/ot-rcp"
-        export OT_CLI_PATH="${OT_BUILDDIR}/posix/${OT_SYSTEM_TRIPLET}/src/posix/ot-cli"
-        export OT_NCP_PATH="${OT_BUILDDIR}/posix/${OT_SYSTEM_TRIPLET}/src/posix/ot-ncp"
+    if [[ "${NODE_MODE}" = 'rcp' ]] ; then
```
Comment (librasungirl) [inline script/test]:
```suggestion
    if [[ "${NODE_MODE}" == 'rcp' ]] ; then
```

File: script/test [hunk 3]
```diff
@@ -192,35 +240,44 @@ do_package() {
...
+
     if [[ -z "$1" ]]; then
         print_usage 1
     fi
 
-    [[ -n "${RADIO_DEVICE}" ]] && echo "Using transceiver mode" || echo "Using standalone mode"
+    [[ "${NODE_MODE}" = "rcp" ]] && echo "Using rcp mode" || echo "Using standalone mode"
     [[ "${NODE_TYPE}" = "ncp-sim" ]] && echo "Using NCP node" || echo "Using CLI node"
     [[ "${VIRTUAL_TIME}" = 1 ]] && echo "Using virtual time" || echo "Using real time"
     [[ "${THREAD_VERSION}" = "1.2" ]] && echo "Using Thread 1.2 stack" || echo "Using Thread 1.1 stack"
```
Comment (librasungirl) [inline script/test]:
```suggestion
    [[ "${NODE_MODE}" == "rcp" ]] && echo "Using rcp mode" || echo "Using standalone mode"
    [[ "${NODE_TYPE}" == "ncp-sim" ]] && echo "Using NCP node" || echo "Using CLI node"
    [[ "${VIRTUAL_TIME}" == 1 ]] && echo "Using virtual time" || echo "Using real time"
    [[ "${THREAD_VERSION}" == "1.2" ]] && echo "Using Thread 1.2 stack" || echo "Using Thread 1.1 stack"
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -42,6 +43,12 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "(Apple)?[Cc]lang")
         set(OT_CFLAGS -Werror)
     endif()
 
+    if(OT_COVERAGE)
+        list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_ENABLE_COVERAGE=1")
+        add_compile_options(-g -O0 --coverage)
```
Comment (jwhui) [inline CMakeLists.txt]: Is this specific to clang? What about gcc?

Comment (bukepo) [reply CMakeLists.txt]: I think this works for gcc as well. https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html#Instrumentation-Options

File: script/test [hunk 5]
```diff
@@ -30,37 +30,89 @@
...
+            options+=("-DOT_SIMULATION_VIRTUAL_TIME_UART=ON")
+        fi
+
+    fi
+
+    local builddir="${OT_BUILDDIR}/cmake/openthread-simulation-${version}"
+    (mkdir -p "${builddir}" \
+        && cd "${builddir}" \
+        && cmake "${OT_SRCDIR}" "${options[@]}" \
+        && make -j"${OT_BUILD_JOBS}")
```
Comment (jwhui) [inline script/test]: How about using ninja?

File: CMakeLists.txt (hunk 4 above)
Comment (jwhui) [reply CMakeLists.txt]: Not clear to me if older versions of gcc (e.g. 4 and 5) support "--coverage", but I'm fine to leave this as is for now.

**Comment by bukepo** [comment]
Ninja is faster:
```bash
# make
./script/test build  65.53s user 13.53s system 753% cpu 10.485 total
# ninja
./script/test build  60.98s user 11.06s system 774% cpu 9.308 total
```

File: script/test (hunk 5 above)
Comment (bukepo) [reply script/test]: done

File: script/bootstrap [hunk 6]
```diff
@@ -59,7 +59,7 @@ install_packages_apt()
 
     # add yapf for pretty
     python3 -m pip install yapf || echo 'Failed to install python code formatter yapf. Install it manually if you need.'
-    
+
```
Comment (bukepo) [inline script/bootstrap]:
```suggestion
```

File: script/test [hunk 7]
```diff
@@ -30,37 +30,89 @@
...
+            options+=("-DOT_SIMULATION_VIRTUAL_TIME_UART=ON")
+        fi
+
+    fi
+
+    local builddir="${OT_BUILDDIR}/cmake/openthread-simulation-${version}"
+    (mkdir -p "${builddir}" \
+        && cd "${builddir}" \
+        && cmake -GNinja "${OT_SRCDIR}" "${options[@]}" \
+        && ninja -j"${OT_BUILD_JOBS}")
```
Comment (jwhui) [inline script/test]: nit: ninja defaults to parallel builds and `-j` is usually not necessary

File: script/test [hunk 8]
```diff
@@ -30,37 +30,89 @@
...
+
+}
+
+build_posix()
+{
+    local version="$1"
+    local options=("${OT_TEST_COMMON_OPTIONS[@]}")
+
+    options+=(
+        "-DOT_PLATFORM=posix-host"
```
Comment (jwhui) [inline script/test]:
```suggestion
        "-DOT_PLATFORM=posix"
```

**Pull request merged**